yt_dlp/extractor/twitter.py

```diff
@@ -816,6 +1049,138 @@ def _real_extract(self, url):
         return info
 
 
+class TwitterSpacesIE(TwitterBaseIE):
+    IE_NAME = 'twitter:spaces'
+    _VALID_URL = TwitterBaseIE._BASE_REGEX + r'i/spaces/(?P<id>[0-9a-zA-Z]{13})'
+    _TWITTER_GRAPHQL = 'https://twitter.com/i/api/graphql/HPEisOmj1epUNLCWTYhUWw/'
+
+    _TESTS = [{
+        'url': 'https://twitter.com/i/spaces/1RDxlgyvNXzJL',
+        'info_dict': {
+            'id': '1RDxlgyvNXzJL',
+            'ext': 'm4a',
+            'title': 'King Carlo e la mossa Kansas City per fare il Grande Centro',
+            'description': 'Twitter Space partecipated by annarita digiorgio, Signor Ernesto, Raffaello Colosimo, Simone M. Sepe',
+            'uploader': r're:Lucio Di Gaetano.*?',
+            'uploader_id': 'luciodigaetano',
+            'live_status': 'was_live',
+            'timestamp': 1659877956397,
+        },
+        'params': {
+            'skip_download': True,  # requires ffmpeg
+        },
+    }, {
+        # Twitter Space not started yet
+        'url': 'https://twitter.com/i/spaces/1mnGeRyljQPJX',
+        'info_dict': {
+            'id': '1mnGeRyljQPJX',
+            'title': '@staderlabs x Hedera AMA w/ @defigirlxoxo, @bmgentile, & @thehbarbull',
+            'description': 'Twitter Space partecipated by nobody yet.',
+            'uploader': 'Hedera',
+            'uploader_id': 'hedera',
+            'live_status': 'is_upcoming',
+            'timestamp': 1662394668924,
+        },
+        'params': {
+            'ignore_no_formats_error': True,
+            'skip_download': True,  # requires ffmpeg
+        },
+        'expected_warnings': ['Twitter Space not started yet', 'No video formats found', 'Requested format is not available'],
+    }]
+
+    def _parse_spaces_data(self, data, space_id):
+        metadata = data.get('metadata')
+        SPACE_SATUS = {
+            'notstarted': 'is_upcoming',
+            'ended': 'was_live',
+            'running': 'is_live',
+            'timedout': 'post_live',
+        }
+        live_status = SPACE_SATUS.get(metadata.get('state').lower())
+
+        # participants
+        participants = join_nonempty(*[p.get('display_name') for p in traverse_obj(data, ('participants', 'speakers'))], delim=', ') or 'nobody yet.'
```

Comment:
```suggestion
        participants = join_nonempty(*traverse_obj(data, ('participants', 'speakers', ..., 'display_name')), delim=', ') or 'nobody yet.'
```
But I wonder if we even have to use `join_nonempty` then, as the `None` should be filtered by `traverse_obj` already?
```suggestion
        participants = ', '.join(traverse_obj(data, ('participants', 'speakers', ..., 'display_name'))) or 'nobody yet.'
```